### ISSUE TITLE
Tensorflow not properly installing

### ISSUE THREAD

**Issue by Bstrum36** [tensorflow/tensorflow#36494]
**System information**
- Windows 10 Pro Version	10.0.17763 Build 17763

- Windows I7 Desktop 16gb memory
- TensorFlow version: 2: (2.1.0)
- Python version: 3.6
- Installed using virtualenv? pip? conda?: installed with latest pip

- CUDA/cuDNN version:CUDA 10.1 ,cudnn64_7.dll
- GPU model and memory: 1080, 16 gb(?)

**Describe the problem**

I have had tensorflow 2. running (no problem) and then attempted to get the gpu version going. No luck I have uninstalled the tensorflow-gpu version and reinstalled (pip install tensorflow), the current version.  

**Installing collected packages: tensorflow
Successfully installed tensorflow-2.1.0**

It appears tensorflow is successfully installed.

Here is my python code: **import tensorflow as tf**
Thats it!

Please advise.  Thanks

**Traceback :**
C:\Users\BillS\AppData\Local\Programs\Python\Python36\python.exe "C:/Users/BillS/AppData/Local/Programs/Python/Python36/Tensorflow test.py"
Traceback (most recent call last):
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow_core\python\pywrap_tensorflow.py", line 58, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow_core\python\pywrap_tensorflow_internal.py", line 28, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow_core\python\pywrap_tensorflow_internal.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\imp.py", line 343, in load_dynamic
    return _load(spec)
ImportError: DLL load failed: The specified module could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:/Users/BillS/AppData/Local/Programs/Python/Python36/Tensorflow test.py", line 1, in <module>
    import tensorflow as tf
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow\__init__.py", line 101, in <module>
    from tensorflow_core import *
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow_core\__init__.py", line 40, in <module>
    from tensorflow.python.tools import module_util as _module_util
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow\__init__.py", line 50, in __getattr__
    module = self._load()
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow\__init__.py", line 44, in _load
    module = _importlib.import_module(self.__name__)
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow_core\python\__init__.py", line 49, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow_core\python\pywrap_tensorflow.py", line 74, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow_core\python\pywrap_tensorflow.py", line 58, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow_core\python\pywrap_tensorflow_internal.py", line 28, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow_core\python\pywrap_tensorflow_internal.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
  File "C:\Users\BillS\AppData\Local\Programs\Python\Python36\lib\imp.py", line 343, in load_dynamic
    return _load(spec)
ImportError: DLL load failed: The specified module could not be found.

Failed to load the native TensorFlow runtime.

See https://www.tensorflow.org/install/errors

for some common reasons and solutions.  Include the entire stack trace
above this error message when asking for help.

Process finished with exit code 1

**Comment by ravikyram** [comment]
@Bstrum36 

What is make/model of your cpu?
I suspect your cpu model does not support AVX instructions sets.See hardware requirements
Make sure to download the latest microsoft visual c++ redistributable from here.
Make sure you update environment path for cuda.Make sure if there is a library that is in a different location/not installed on your system that cannot be loaded.Also, please follow the instructions from Tensorflow website.
Please, check Your CPU/Python is on 32 bits?Please, refer #36167 and see if it helps you. Thanks!

**Comment by Bstrum36** [comment]
Thanks Ravikyram,

I just installed the the latest Visual Studio from your supplied link. That didn have any effect.
A to CPU  Its an Intel I&-7700K 4.2 ghz.  I have 16 Gb memory on my PC and am running 64 bit Windows.  

What should the environment path be for cuda?  Her is the path.  C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA. Should I reset this?

How do  tell about AVX instructions? Tensorflow 2. worked before I treid to install the gpu version.

Thanks

**Comment by ravikyram** [comment]
@Bstrum36 

For AVX instructions please go through the [link](https://en.wikipedia.org/wiki/Advanced_Vector_Extensions#CPUs_with_AVX) .To set to set cuda path , please refer this https://www.tensorflow.org/install/gpu#linux_setup . Thanks!

**Comment by Bstrum36** [comment]
Thanks again, No luck.

Here are the paths copied directly for the environment variable tab:
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\bin
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\libnvvp
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\extras\CUPTI\lib64
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\include
C:\tools\cuda\bin

The libnvxp entry was already there.
I rebooted and still got exactly the the same errors.
Also I check my CPU and it is listed as AVX enabled.

I have been successfully using tensor flow 2.0  the problem occured when trying to get the GP working by unistall ver 2.o and installing the - gpu version.
I gave up on the gpu and uninstalled the version and have reinstalled 2.0 without errors. 

I suppose I could do a system restore and hope for the best but I'd rather not go through that mess.

Any more ideas on how to proceed??
